packages/next/src/client/components/router-reducer/router-reducer-types.ts

```diff
@@ -117,6 +117,14 @@ export interface ServerPatchAction {
   mutable: Mutable
 }
 
+/**
+ * PrefetchKind defines the type of prefetching that should be done.
+ * - `auto` - if the page is dynamic, prefetch the page data partially, if static prefetch the page data fully.
+ * - `full` - prefetch the page data fully.
+ * - `temporary` - a temporary prefetch entry is added to the cache, this is used when prefetch={false} is used in next/link or when you push a route programmatically.
+ */
+export type PrefetchKind = 'auto' | 'full' | 'temporary'
```

Comment:
Can you make sure these use a similar approach as the reducer actions, otherwise we'll run into problems down the line when the code is refactored, also makes it easier to search for usage of it.

https://github.com/vercel/next.js/blob/189e6a3687924de3172438a2912973bd49c1e37e/packages/next/src/client/components/router-reducer/router-reducer-types.ts#L9